packages/react-examples/src/react-charting/DeclarativeChart/ErrorBoundary.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+
+interface ErrorBoundaryProps {
+  children: React.ReactNode;
+}
+
+interface ErrorBoundaryState {
+  hasError: boolean;
+  error: string;
+}
+
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false, error: '' };
+  }
+
+  static getDerivedStateFromError(error: Error) {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true, error: `${error.message} ${error.stack}` };
+  }
+
+  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
+    console.error("Error caught by ErrorBoundary:", error, errorInfo);
+  }
+
+  render() {
+    if (this.state.hasError) {
+      return <h1>${this.state.error}</h1>;
```

Comment:
will we be showing the entire error stack on screen? instead can we console log the stack and show some message on screen?

Reply:
console log is not recommended. This is a test example only. It is fine to show the error on the screen.